aws/resource_aws_default_network_acl.go

```diff
@@ -64,32 +68,41 @@ func resourceAwsDefaultNetworkAcl() *schema.Resource {
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"from_port": {
-							Type:     schema.TypeInt,
-							Required: true,
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IsPortNumberOrZero,
 						},
 						"to_port": {
-							Type:     schema.TypeInt,
-							Required: true,
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IsPortNumberOrZero,
 						},
 						"rule_no": {
-							Type:     schema.TypeInt,
-							Required: true,
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validation.IntBetween(1, 32766),
```

Comment:
Nit: Doesn't need to be this PR, but we should probably create a constant for this, e.g.

```go
const (
	// Maximum number for EC2 Network ACL Rules. The range 32767 to 65535 is reserved for internal use.
	Ec2NetworkAclRuleNumberMaximum = 32766
)
```

We'll need to figure out the best place for these in the future when things are broken into separate packages.